RecoLocalTracker/SiPhase2Clusterizer/plugins/Phase2TrackerClusterizer.cc

```diff
@@ -0,0 +1,89 @@
+#include "RecoLocalTracker/SiPhase2Clusterizer/plugins/Phase2TrackerClusterizer.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDetUnit.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include <vector>
+#include <iostream>
+
+namespace cms {
+
+    /*
+     * Initialise the producer
+     */ 
+
+    Phase2TrackerClusterizer::Phase2TrackerClusterizer(edm::ParameterSet const& conf) :
+        conf_(conf),
+        maxClusterSize_(conf.getParameter< unsigned int >("maxClusterSize")),
+        maxNumberClusters_(conf.getParameter< unsigned int >("maxNumberClusters")) {
+            // Objects that will be used
+            token_ = consumes< edm::DetSetVector< Phase2TrackerDigi > >(conf.getParameter<edm::InputTag>("src"));
+            // Objects that will be produced
+            produces< Phase2TrackerCluster1DCollectionNew >(); 
+            // 
+            clusterizer_  = new Phase2TrackerClusterizerAlgorithm(maxClusterSize_, maxNumberClusters_);
```

Comment:
use std::unique_ptr instead (no need to delete in this case).
token_ and clusterizer_ could be initialized in the initializer list (between ":" and "{" above.